firetower

```diff
@@ -195,24 +195,29 @@ function get_flag() {
 
 function should_preserve_scrollback() {
   for var in "$@" ; do
-    if [[ $var == --preserve-scrollback ]] ; then
+    if [[ $var == --preserve-scrollback ]] || [[ $var == -p ]] ; then
       return 0
     fi
   done
   return 1
 }
 
 function get_directory() {
-  for var in "$@"
-  do
-    if [[ $var == --directory=* ]] ; then
-      echo "$var" | cut -d '=' -f 2
-      return
-    fi
+  for (( i=1; i<=$#; i++)); do
+    j=$((i+1))
+     if ( [[ ${!i} == -d ]] && [[ ${!j} == * ]] ) ; then
+	echo "${!j}"
+	return
+     elif ( [[ ${!i} == --directory=* ]] ) ; then
+	echo "${!i}" | cut -d '=' -f 2
+	return
+     fi
   done
   echo "./"
 }
 
+
+
```

Comment:
When you add files, if you add with `git add -p [filename]` instead of `git add filename`, you can review each individual edit and `add`, `skip`, etc.. So you won't accidentally commit empty lines like this.